src/snowflake/connector/auth.py

```diff
@@ -517,6 +571,34 @@ def write_temporary_credentials(self, host, user, session_parameters, response):
             )
         return
 
+    # Default timeout handler. This will trigger if the authenticator
+    # hasn't implemented one. By default we retry on timeouts and use
+    # jitter to deduce the time to sleep before retrying. The sleep
+    # time ranges between 1 and 16 seconds.
+    def handle_timeout(
+        self,
+        authenticator: str,
+        service_name: any,
+        account: str,
+        user: str,
+        password: str,
+    ) -> None:
+        del authenticator, service_name, account, user, password
+        if not self._retry_ctx.should_retry():
+            raise OperationalError(
+                msg="Could not connect to Snowflake backend after {} attempt(s)."
+                "Aborting".format(self._retry_ctx.get_current_retry_count()),
+                errno=ER_FAILED_TO_CONNECT_TO_DB,
+            )
+        else:
+            logger.debug(
+                "Hit connection timeout, attempt number {}. Will retry in a bit...".format(
+                    self._retry_ctx.get_current_retry_count()
```

Comment:
I find f-strings not only easier to read, but from my testing they seem to be faster too:
```pycon
>>> import timeit
>>> timeit.timeit('"some string: {}".format(\'asd\')')
0.16692085900285747
>>> timeit.timeit('f"some string: {\'asd\'}"')
0.05588944599730894
```